### DISCUSSION TITLE
Enable or disable a task using a global param, without a default branch

### DISCUSSION THREAD

**Discussion by frouhi** [apache/dolphinscheduler#12426] [Q&A]
Currently, the only way we can enable/disable a specific task via a global param is using a switch. However, a switch requires a default branch. My current workaround is having a dummy task as a default, but this makes the DAG unnecessary cluttered. Is there any way to not have a default for the switch? (I'm generating my DAGs using python)

**Comment by DarkAssassinator** [comment]
did u mean that u do not want to choose any sub task?

**Reply by frouhi** [reply]
I want to have only one sub task, which either runs or doesn't, depending on the global param condition (currently a default task seems to be required when using a switch, so we must have 2 sub tasks).

**Reply by DarkAssassinator** [reply]
> I want to have only one sub task, which either runs or doesn't, depending on the global param condition (currently a default task seems to be required when using a switch, so we must have 2 sub tasks).

Hi @frouhi  does Dependent task meet ur need?

**Reply by frouhi** [reply]
Not really! I don't think Dependent tasks support a condition on global params! For example, I need something like this: condition="${var} > 1". I don't see a condition arg provided by Dependent task! But please let me know if I'm missing something from the docs.

**Reply by DarkAssassinator** [reply]
> Not really! I don't think Dependent tasks support a condition on global params! For example, I need something like this: condition="${var} > 1". I don't see a condition arg provided by Dependent task! But please let me know if I'm missing something from the docs.

as i know, current ds do not support, but i think that this is a good feature, could u please create a issue? if the community approves, we can develop this feature

**Reply by frouhi** [reply]
Sure, I'll open an issue.
thanks
